src/main/java/org/jitsi/videobridge/cc/BitrateController.java

```diff
@@ -1346,37 +1315,42 @@ private TrackBitrateAllocation(
             long idealBps = 0;
             for (RTPEncodingDesc encoding : encodings)
             {
-                if (maxFrameHeight >= 0 && encoding.getHeight() > maxFrameHeight)
+                int idealHeight = videoConstraints.getIdealHeight();
+                // NOTE that this condition is a bit too rigid: Imagine an
+                // endpoint without simulcast where it sends a single 720p track
+                // with a single encoding and imagine that the ideal resolution
+                // is 180p for that endpoint because it's a thumbnail.
+                //
+                // The guard below should lead to no video for that endpoint.
+                //
+                // It works today because of how we interpret signaling from the
+                // endpoints: the bridge assumes that the first encoding is 180p
+                // (and that the second is 360p and the third 720p).
+                //
+                // When we change how tracks are signaled or how we discover the
+                // resolution of each individual encoding, this guard will have
+                // to be tweaked.
+                if (idealHeight >= 0 && encoding.getHeight() > idealHeight)
                 {
                     continue;
                 }
-                if (selected)
-                {
-                    // For the selected participant we favor frame rate over
-                    // resolution. Basically what we want for the on-stage
-                    // participant is 180p7.5fps, 180p15fps, 180p30fps,
-                    // 360p30fps and 720p30fps.
-                    if (encoding.getHeight() < Config.onstagePreferredHeightPx()
-                        || encoding.getFrameRate() >= Config.onstagePreferredFramerate())
-                    {
-                        long encodingBitrateBps = encoding.getBitrateBps(nowMs);
-                        if (encodingBitrateBps > 0)
-                        {
-                            idealBps = encodingBitrateBps;
-                        }
-                        ratesList.add(
-                            new RateSnapshot(encodingBitrateBps, encoding));
-                    }
 
-                    if (encoding.getHeight() <= Config.onstagePreferredHeightPx())
-                    {
-                        ratedPreferredIdx = ratesList.size() - 1;
-                    }
-                }
-                else if (encoding.getHeight() <= Config.thumbnailMaxHeightPx())
+                // For the "selected" participant we favor frame rate over
+                // resolution. Basically what we want for the on-stage
+                // participant is 180p7.5fps, 180p15fps, 180p30fps, 360p30fps
+                // and 720p30fps. For the thumbnails, we consider all temporal
+                // layers of the low resolution stream.
+
```

Comment:
Does this generalize?  Can it be described in terms of the generalized video constraints, with this common case as an example?